src/modules/map/components/LocationPopupShare.vue

```diff
@@ -0,0 +1,122 @@
+<script setup>
+import { computed, onMounted, ref, toRefs, watch } from 'vue'
+import { useRoute } from 'vue-router'
+import { useStore } from 'vuex'
+
+import { createShortLink } from '@/api/shortlink.api'
+import MenuShareInputCopyButton from '@/modules/menu/components/share/MenuShareInputCopyButton.vue'
+import MenuShareSocialNetworks from '@/modules/menu/components/share/MenuShareSocialNetworks.vue'
+import { stringifyQuery } from '@/utils/url-router'
+
+const store = useStore()
+const emit = defineEmits(['shareLink'])
+
+const props = defineProps({
+    coordinate: {
+        type: Boolean,
+        required: true,
+    },
+    clickInfo: {
+        type: Object,
+        required: true,
+    },
+    currentLang: {
+        type: Object,
+        required: true,
+    },
+    showEmbedSharing: {
+        type: Boolean,
+        required: true,
+    },
+})
+const { coordinate, clickInfo, currentLang, showEmbedSharing } = toRefs(props)
+const shareLinkUrlShorten = ref(null)
+const shareLinkUrl = ref(null)
+const route = useRoute()
+
+const zoom = computed(() => store.state.position.zoom)
+
+onMounted(() => {
+    if (clickInfo.value) {
+        if (showEmbedSharing.value) {
+            updateShareLink()
+        }
+    }
+})
+
+watch(clickInfo, () => {
+    if (showEmbedSharing.value) {
+        updateShareLink()
+    }
+})
+watch(currentLang, () => {
+    if (showEmbedSharing.value) {
+        setTimeout(() => updateShareLink(), 1)
```

Comment:
Why the `setTimeout` here ? if you have a race condition it could be also solved using vue `nextTick` method